docs/dev/relay_op_strategy.rst

```diff
@@ -0,0 +1,256 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+.. _relay-op-strategy:
+
+Relay Operator Strategy
+=======================
+
+In order to lower Relay operators to implementation defined in TOPI library, the
+compute and schedule functions need to be registered to Relay operators.
+However, compute and schedule functions are usually specialized for each target,
+and further, even for the same target, we may have multiple algorithms and
+implementations available. To deal with the complexity, we introduce operator
+strategy to allow developers to define a flexible lowering strategy for each
+operator and target.
+
+
+Operator Strategy Design
+------------------------
+
+The basic element in operator strategy is an ``OpImplementation``. It includes
+the a pair of compute and schedule function, the name of the implementation,
+and a priority level (the usability of priority level will be explained below).
+
+The ``OpStrategy`` includes a list of specializations. Each specialization
+contains a list of ``OpImplementation`` associated with a specialized condition
+(see ``SpecializedCondition`` definition in ``include/tvm/te/schedule.h``).  The
+specialized condition can be null, indicating the implementations are generally
+applicable; otherwise, the implementations should only be used when the
+specialized condition is satisfied. ``OpStrategy`` provides only one API,
+adding an implementation to the strategy:
+
+.. code:: python
+
+    def add_implementation(self, compute, schedule, name="default", plevel=10)
```

Comment:
The API here seems a bit confuse. Specifically, it doesn't use any keywords mentioned in the above paragraph. My feeling is this section focuses on components on C++ backend, so would that be better to move this API intro to the next section along with other Python interfaces?

Reply:
Moved to the later part.